dev/Scroller/Scroller.cpp

```diff
@@ -1757,11 +1776,31 @@ void Scroller::EnsureTransformExpressionAnimations()
     }
 }
 
-void Scroller::SetupSnapPoints(std::set<winrt::ScrollerSnapPointBase, winrtProjectionComparator>* snapPoints, ScrollerDimension dimension)
+void Scroller::SetupSnapPoints(
+    _In_opt_ std::set<winrt::ScrollSnapPointBase, winrtProjectionComparator>* scrollSnapPointsSet,
+    _In_opt_ std::set<winrt::ZoomSnapPointBase, winrtProjectionComparator>* zoomSnapPointsSet,
+    ScrollerDimension dimension)
 {
     MUX_ASSERT(!SharedHelpers::IsTH2OrLower());
+    MUX_ASSERT(scrollSnapPointsSet || zoomSnapPointsSet);
+    MUX_ASSERT(!(scrollSnapPointsSet && zoomSnapPointsSet));
```

Comment:
I think you can avoid the use of this pattern by utilizing generic functions, per in person talk with Jevan.

Reply:
Yes. Done.